pkg/patterns/declarative/pkg/applier/direct.go

```diff
@@ -128,16 +128,19 @@ func (d *DirectApplier) Apply(ctx context.Context, opt ApplierOptions) error {
 	for i, arg := range opt.ExtraArgs {
 		switch arg {
 		case "--force":
+			// TODO Does this do anything? It seems like opt (aka ApplierOptions) is not used anymore
 			opt.Force = true
 		case "--prune":
 			applyOpts.Prune = true
 		case "--selector":
 			applyOpts.Selector = opt.ExtraArgs[i+1]
+		case "--prune-whitelist":
+			applyOpts.PruneWhitelist = append(applyOpts.PruneWhitelist, opt.ExtraArgs[i+1])
```

Comment:
Suggestion (non-blocking): You might consider exposing a strongly-typed form like `PruneWhitelist []schema.GroupVersionKind` or `[]schema.GroupKind`, more like what we're doing with `Force`

Reply:
I needed to use `applyOpts.PruneResources` anyways. `PruneWhitelist` did not seem to have the desired (or any) effect.